src/instructlab/model/upload.py

```diff
@@ -276,3 +298,91 @@ def upload_oci(self) -> None:
         logger.info(
             f"Uploading OCI model at {self.local_model_path} succeeded!",
         )
+
+
+class S3ModelUploader(ModelUploader):
+    """Class to handle uploading models to S3"""
+
+    def __init__(
+        self,
+        model: str,
+        destination: str,
+    ) -> None:
+        super().__init__(
+            model=model,
+            destination=destination,
+        )
+
+    def upload(self) -> None:
+        """
+        Upload specified model to the specified S3 endpoint
+        """
+
+        # AWS cred check
+        try:
+            sts_client = boto3.client("sts")
+            sts_client.get_caller_identity()
+        except Exception as exc:
+            raise ValueError(
+                """AWS credentials need to be set in your environment to upload to S3.
+                You can set your credentials using the AWS CLI
+                For alternative methods to set credentials, please review the boto3 documentation"""
+            ) from exc
```

Comment:
So... there are many reasons one may not be able to authenticate with AWS, one of them being unset credentials, but another (for example) being invalid/expired credentials. (Some people have credentials automatically rotated, sometimes as often as every 30 mins.)

Also, the recommended approach is to use an AWS credentials file to make the credentials automatically load.

```suggestion
            raise ValueError(
                """AWS credentials need to be set in your environment to upload to S3. 
                If you have not already set your credentials, you can set them using the 
                AWS CLI or through an AWS credentials file. 
                If you have set your credentials, please ensure they are valid and not expired.
                For alternative methods to set credentials, please review the boto3 documentation."""
            ) from exc
```

Reply:
Done!